ecs-cli/modules/cli/compose/project/project.go

```diff
@@ -134,11 +134,25 @@ func (p *ecsProject) Parse() error {
 // parseCompose sets data from the compose files on the ecsProject
 func (p *ecsProject) parseCompose() error {
 	logrus.Debug("Parsing the compose yaml...")
+
+	// check for Compose version and call appropriate parsing function
+	version, err := p.checkComposeVersion()
+	if err != nil {
+		return err
+	}
+	switch version {
+	case "", "1", "1.0", "2", "2.0":
```

Comment:
You might consider using `strconv.ParseFloat` instead of doing a string match here, eg:
```
	v, _ := strconv.ParseFloat(version, 32)
	if v == 1 || v == 2 {
		parseV1V2
	}
```

Or something like that.

Reply:
Why not just check if the string starts with "2" or "3"?

Reply:
If you hard code the exact strings, then we are stuck supporting those version strings only (related: https://github.com/aws/amazon-ecs-cli/issues/305).

Reply:
I like @SoManyHs idea of parsing into a float, but does Compose use semantic versioning? "2.1.5" is a valid version but not a valid float.

Reply:
Looks like compose using "x.x" versioning: https://docs.docker.com/compose/compose-file/compose-versioning/#compatibility-matrix, so `strconv` might still be alright. I am wondering though if we were only accounting for 3.0 vs the minor versions of compose 3, since there do appear to be other fields.

Reply:
re: versioning - Currently, libcompose throws an unmarshalling error if minor versions are detected (e.g., 2.1), so scoping 3 support the major version "3" or "3.0" would be consistent (similarly, Kompose does not support minor versions of 3). 

However, I'm all for supporting up to what is feasible up to 3.6 (latest). I suggest we start with supporting 3/3.0 and work on supporting minor versions of 3 in a future iteration(s). Thoughts?

Reply:
Why not just build it to support all versions now? Libcompose could fix their stuff so that minor versions can be unmarshalled, and if we support minor versions too, then we aren't leaving around any future work that needs to be done.

Reply:
For now, scoping the parsing to support just the major versions seems most feasible. We've already learned we can't rely on libcompose for support of different versions, and to date it doesn't seem like we've had many customers ask for minor versions of v2. As I understand it, the ask by customers for v3 was just for major versions. Limiting the scope to major releases is probably our best bet for moving forward on this.

Based on that, the original implementation should be fine. My suggestion for using `strconv` was mostly to consider edge cases and minor versions, but the latter may now be irrelevant. Including unit tests will also help with this :)

Reply:
@PettitWesley my comment mainly pertains to the scope of this code review. The work needed to differentiate between features supported in different minor versions of 3 will be better understood after we start implementing parsing and task def conversion, which hasn't been started yet. I'd prefer to get basic support for v3 working and then work on supporting minor versions after we better understand what that would entail.

+1 to @SoManyHs  comment re: unit tests.